From ac9e62a27c81233d0a2550736b14bc0872b7d477 Mon Sep 17 00:00:00 2001
From: Claudiu Zissulescu <claziss@gmail.com>
Date: Mon, 8 Apr 2019 19:39:33 +0300
Subject: [PATCH 7/7] [ARC][PART3] Add ARCHS release 310a tune variant.

---
 gcc/config/arc/arc.c | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/gcc/config/arc/arc.c b/gcc/config/arc/arc.c
index 6cec1bfd227..78a37aa2a18 100644
--- a/gcc/config/arc/arc.c
+++ b/gcc/config/arc/arc.c
@@ -7713,6 +7713,21 @@ arc_hazard (rtx_insn *pred, rtx_insn *succ)
   if (TARGET_ARC600)
     return arc600_corereg_hazard (pred, succ);
 
+  return 0;
+}
+
+static int
+arc_check_release31a (rtx_insn *pred, rtx_insn *succ)
+{
+  if (!pred || !INSN_P (pred) || !succ || !INSN_P (succ))
+    return 0;
+
+  if (!JUMP_P (pred) && !single_set (pred))
+    return 0;
+
+  if (!JUMP_P (succ) && !single_set (succ))
+    return 0;
+
   if (TARGET_HS && (arc_tune == ARC_TUNE_ARCHS4X_REL31A))
     switch (get_attr_type (pred))
       {
@@ -7724,7 +7739,7 @@ arc_hazard (rtx_insn *pred, rtx_insn *succ)
 	  case TYPE_LOOP_END:
 	    return 1;
 	  default:
-	   break;
+	    break;
 	  }
 	break;
       case TYPE_BRCC:
@@ -7754,10 +7769,8 @@ workaround_arc_anomaly (void)
   for (insn = get_insns (); insn; insn = NEXT_INSN (insn))
     {
       succ0 = next_real_insn (insn);
-      if (arc_hazard (insn, succ0))
-	{
-	  emit_insn_before (gen_nopv (), succ0);
-	}
+      if (arc_hazard (insn, succ0) || arc_check_release31a (insn, succ0))
+	emit_insn_before (gen_nopv (), succ0);
     }
 
   if (!TARGET_ARC700)
-- 
2.16.2

