--- make-orig/glob/glob.c	2015-02-12 20:29:23.373816594 +0300
+++ make-patched/glob/glob.c	2015-02-12 20:29:04.226535073 +0300
@@ -178,8 +178,10 @@
 
 #if !defined HAVE_MEMPCPY && __GLIBC__ - 0 == 2 && __GLIBC_MINOR__ >= 1
 # define HAVE_MEMPCPY	1
-# undef  mempcpy
-# define mempcpy(Dest, Src, Len) __mempcpy (Dest, Src, Len)
+# ifndef __UCLIBC__
+#  undef  mempcpy
+#  define mempcpy(Dest, Src, Len) __mempcpy (Dest, Src, Len)
+# endif
 #endif
 
 #if !defined __GNU_LIBRARY__ && !defined __DJGPP__
@@ -207,9 +209,7 @@
 # endif /* __SASC */
 #endif /* __GNU_LIBRARY__ || __DJGPP__ */
 
-
-#if !defined __alloca && !defined __GNU_LIBRARY__
-
+#if !defined __alloca && (!defined __GNU_LIBRARY__ || defined __UCLIBC__)
 # ifdef	__GNUC__
 #  undef alloca
 #  define alloca(n)	__builtin_alloca (n)
@@ -231,6 +231,10 @@
 
 #endif
 
+#ifdef __UCLIBC__
+# define __stat stat
+#endif
+
 #ifndef __GNU_LIBRARY__
 # define __stat stat
 # ifdef STAT_MACROS_BROKEN
